scripts/lib/compile-scss-with-variables.js

```diff
@@ -0,0 +1,187 @@
+const sass = require('sass-embedded');
+const { pathToFileURL } = require('url');
+const path = require('path');
+
+/* The markers are chosen as symbols that would never be used as variable names under normal
+ * circumstances. They also get wrapped in "--" to reduce the chances of collision with an actual variable.
+ * The markers could have been dynamically generated after inspecting all of the variables but it is not
+ * worth the performance penalty.
+ *
+ * Note: The `varListMarker` has to start with a colon or period to avoid Sass errors.
+ */
+const varListMarker = ':֍෴֍߷';
+const arrayMarker = '෴';
+const objectMarker = '߷';
+const forbiddenKeyNames = [...Object.getOwnPropertyNames(Object.prototype), 'prototype'];
```

Comment:
Since they are local variables and not exported, I didn't; do you have a strong feeling towards this?
PS, for the same reason, i didn't `deepFreeze` the forbidden key names.